app/controllers/assumed/ViewAssumedReportsController.scala

```diff
@@ -17,40 +17,75 @@
 import queries.AssumedReportSummariesQuery
 import repositories.SessionRepository
 import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
 import views.html.assumed.ViewAssumedReportsView
 
+import java.time.{Clock, Year}
 import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 
 class ViewAssumedReportsController @Inject()(override val messagesApi: MessagesApi,
                                              identify: IdentifierAction,
                                              val controllerComponents: MessagesControllerComponents,
-                                             connector: AssumedReportingConnector,
+                                             assumedReportingConnector: AssumedReportingConnector,
+                                             platformOperatorConnector: PlatformOperatorConnector,
                                              sessionRepository: SessionRepository,
-                                             view: ViewAssumedReportsView)
+                                             view: ViewAssumedReportsView,
+                                             formProvider: ViewAssumedReportsFormProvider,
+                                             clock: Clock)
                                             (implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
 
-  def onPageLoad(): Action[AnyContent] = identify.async {
-    implicit request =>
-      connector.list.flatMap { submissions =>
-
-        submissions.groupBy(_.operatorId).toList
-          .traverse { (operatorId, submissions) =>
-            for {
-              answers <- Future.fromTry(UserAnswers(request.userId, operatorId).set(AssumedReportSummariesQuery, submissions))
-              _       <- sessionRepository.set(answers)
-            } yield Done
-          }
-          .map(_ => Ok(view(submissions)))
+  def onPageLoad(): Action[AnyContent] = identify.async { implicit request =>
+    formProvider().bindFromRequest().fold(
+      _ => Future.successful(Redirect(baseRoutes.JourneyRecoveryController.onPageLoad())),
+      formData => {
+        (for {
+          assumedReports <- assumedReportingConnector.list
+          filteredAssumedReports = filterBy(assumedReports, formData)
+          operatorsResponse <- platformOperatorConnector.viewPlatformOperators
+        } yield {
+          assumedReports.groupBy(_.operatorId).toList.traverse { (operatorId, submissions) =>
+              for {
+                answers <- Future.fromTry(UserAnswers(request.userId, operatorId).set(AssumedReportSummariesQuery, submissions))
+                _ <- sessionRepository.set(answers)
+              } yield answers
+            }
+            .map(_ => {
+              val viewModel = ViewAssumedReportsViewModel(
+                platformOperators = operatorsResponse.platformOperators,
+                assumedReportingSubmissionSummaries = filteredAssumedReports,
```

Comment:
Add to test for when a reporting period is selected in the form data

Reply:
Test updated.